File: mslibrary-kotlindocs/src/main/java/kr/msleague/msgui/api/event/MSGuiEvent.java
package kr.msleague.msgui.api.event;

import kr.msleague.msgui.gui.MSGui;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;

@AllArgsConstructor
@Getter
public abstract class MSGuiEvent extends Event {
    protected Player player;
    protected MSGui<?> gui;
}

File: mslibrary-gui/huni-gui/src/main/java/kr/msleague/msgui/api/events/MSGuiEvent.kt
package kr.msleague.msgui.api.events

import kr.msleague.msgui.api.interfaces.MSGui
import kr.msleague.msgui.api.interfaces.MSGuiViewer
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

abstract class MSGuiEvent(
    val viewer: MSGuiViewer,
    val gui: MSGui
): Event() {

    override fun getHandlers(): HandlerList = HANDLER_LIST

    companion object {
        var HANDLER_LIST = HandlerList()
        @JvmStatic fun getHandlerList(): HandlerList = HANDLER_LIST
    }

}